chainer/variable.py

```diff
@@ -843,16 +846,18 @@ def array(self, d):
 
         if self.xp is chainerx:
             d_old = self._data[0]
+            assert d_old is None or isinstance(d_old, chainerx.ndarray)
             if (d_old is not None
-                    and (d_old.is_backprop_required()  # type: ignore
+                    and (d_old.is_backprop_required()
                          or d.is_backprop_required())):
```

Comment:
I'm very sorry that the signature of this method should be `(types.NdArray) -> None`... I fixed it in #6016.